plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2060,6 +2076,23 @@ public void renameMaterializedView(ConnectorSession session, SchemaTableName sou
         catalog.renameMaterializedView(session, source, target);
     }
 
+    // Information that should be added to every generated Iceberg snapshot, in its summary.
+    // TODO - Consider moving this to the catalog module (on commit).
+    public static Map<String, String> getExtraSummaryMetadata(ConnectorSession session, Optional<String> trinoVersion)
+    {
+        ImmutableMap.Builder<String, String> snapshotSummary = ImmutableMap.<String, String>builder()
+                .put(TRINO_QUERY_ID_NAME, session.getQueryId());
+
+        trinoVersion.map(version -> snapshotSummary.put(TRINO_VERSION, version));
+
+        return snapshotSummary.buildOrThrow();
+    }
+
+    public static Map<String, String> getExtraSummaryMetadata(ConnectorSession session)
+    {
+        return getExtraSummaryMetadata(session, Optional.empty());
+    }
```

Comment:
This presently does _not_ use the `trinoVersion`. I could not figure out how to get it from within the `IcebergMetadata`.

Possibly I can move this function over to `AbstractTrinoCatalog` and then it would be available?

Reply:
> This presently does _not_ use the `trinoVersion`. I could not figure out how to get it from within the `IcebergMetadata`.

see `HiveMetadataFactory` for example.

the value comes from Guice and should be injected wherever it is needed.

Reply:
Cool. Thank you! I'm not super familiar with Guice but I'll take a look at `HiveMetadataFactory`.